sdcm/provision/helpers/cloud_init.py

```diff
@@ -31,12 +34,24 @@ def wait_cloud_init_completes(remoter: RemoteCmdRunnerBase, instance: VmInstance
     LOGGER.info("Waiting for cloud-init to complete on node %s...", instance.name)
     errors_found = False
     remoter.is_up(60 * 5)
-    result = remoter.sudo("cloud-init status --wait")
-    status = result.stdout
-    LOGGER.debug("cloud-init status: %s", status)
-    if "done" not in status:
-        LOGGER.error("Some error during cloud-init %s", status)
-        errors_found = True
+    # examples: 24.1.3-0ubuntu3.3, 19.3-46.amzn2.0.2
+    cloud_init_version = Version(remoter.sudo("cloud-init --version 2>&1").stdout.split()[1].split('-')[0])
+    # cloud-init supports json output from version 23.4, see:
+    # https://cloudinit.readthedocs.io/en/latest/explanation/return_codes.html#id1
+    if cloud_init_version > Version("23.4"):
+        result = remoter.sudo("cloud-init status --format=json --wait", ignore_status=True)
+        status = json.loads(result.stdout)
+
+        LOGGER.debug("cloud-init status: %s", status)
+        if not status['status'] == "done" or status['errors'] or result.return_code == 1:
```

Comment:
The following code:
```
not status['status'] == "done"
```

should be replaced with the following:

```
status['status'] != "done"
```

Reply:
changed